Src/BootCamp.Chapter/LinqMethodDemo.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    public static class LinqMethodDemo
+    {
+        internal static IEnumerable<T> Demo<T>(this IEnumerable<T> list, Predicate<T> predicate)
+        {
+            list = Filter(list, predicate);
```

Comment:
Both methods are not needed and all the implementation can be moved to this methods (without private any private step functions)

Reply:
so I do not lt you see that the extensions are working or do I misunderstood you here complete

Reply:
Refer to comments below.